libraries/preferences/api/src/main/kotlin/io/element/android/features/preferences/api/store/SessionPreferencesStore.kt

```diff
@@ -19,8 +19,20 @@ package io.element.android.features.preferences.api.store
 import kotlinx.coroutines.flow.Flow
 
 interface SessionPreferencesStore {
+    suspend fun setSharePresence(enabled: Boolean)
+    fun isSharePresenceEnabled(): Flow<Boolean>
+
     suspend fun setSendPublicReadReceipts(enabled: Boolean)
     fun isSendPublicReadReceiptsEnabled(): Flow<Boolean>
```

Comment:
The internal key should probably be kept for migration, but maybe we should remove the public methods from the interface here? I see it seems to still be used in `TimelinePresenter`, but maybe should we use `sharePresence` there instead?

Reply:
I hadn't thought about forks, but that makes sense, thanks! I still think this is more complicated than it might need to be, but if a decision was made around this then it LGTM.

Reply:
> I believe forks (and maybe future us) will not want to use `setSharePresence` but may want to add more specific settings.

I mentioned this in #element-x-android:matrix.org but as a user, I would definitely like to see this split into two options:
* share and display read receipts
* share and display typing notifications

The way I described it in the EXA room is that I don't want my presence shared *passively* (read receipts) but I'm happy for my presence to be shared *actively* (typing notifications). For me, the former is uncomfortable over-sharing, while the latter is a sensible and useful result of consciously interacting with the room. My feelings about each are the complete opposite of one another, so combining them into a single setting forces me to compromise one one of them.

Combining sharing and displaying of each into a single option doesn't bother me; I have the same feelings about other people's presence notifications as I do about my own, i.e. I don't want to see *your* read receipts but I do think *your* typing notifications are extremely helpful. It's a very informal poll, but when this was discussed on Matrix, based on messages and emoji reactions, ~3 users agreed with this position (supporting combined sharing/display into a single setting) and 1 disagreed (preferring separate settings for sharing and display).

Personally, I feel like *four* different settings in the UI relating to presence is a bit bloated for what is shaping up to be the "default" Matrix client on Android, but combining them all into *one* is overly broad.